### PR TITLE
wr-129: implement sign in with google

### PR DESCRIPTION
**Pull request by canterbery**
- Add Sign In with Google button on Sign In page
-  sign in with google should open default Google OAuth window
-  Upon successful Google login - make BE request to sign user in with our credentials

### CODE REVIEW

**Review by NazarKovtun1 — DISMISSED** [review]

File: backend/.env.example [hunk 1]
```diff
@@ -31,3 +31,8 @@ JWT_SECRET_KEY=secret
 # OPEN_AI
 #
 OPEN_AI_API_KEY=""
+#
+# GOOGLE AUTH
+#
+CLIENT_ID=""
+CLIENT_SECRET=""
```
Comment (andanf-e) [inline backend/.env.example]: To make it more clear
```suggestion
GOOGLE_CLIENT_ID=""
GOOGLE_CLIENT_SECRET=""
```

File: frontend/src/pages/auth/components/sign-in-form/styles.module.scss [hunk 2]
```diff
@@ -52,3 +52,12 @@
...
   font-family: "Libre Baskerville", sans-serif;
 }
+
+.or {
+  align-self: center;
+  margin-top: 20px;
+  color: #b6b6b6;
+  font-weight: 500;
+  font-size: 18px;
+  font-family: Montserat, sans-serif;
```
Comment (andanf-e) [inline frontend/src/pages/auth/components/sign-in-form/styles.module.scss]: Let's use Raleway (Montserrat is used on design, but 4 fonts I guess is to much :c )

File: frontend/src/pages/auth/components/auth-sign-in-button/auth-sign-in-button.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { type ButtonProperties } from '~/libs/components/button/button.js';
+import { Button } from '~/libs/components/components.js';
+
+import styles from './styles.module.scss';
+
+type Properties = Pick<ButtonProperties, 'label' | 'disabled' | 'onClick'>;
+
+const AuthSignInButton: React.FC<Properties> = (properties) => {
+  return (
+    <Button {...properties} className={styles.authSignInButton} type="submit" />
```
Comment (andanf-e) [inline frontend/src/pages/auth/components/auth-sign-in-button/auth-sign-in-button.tsx]: Please pass properties explicitly

File: frontend/src/pages/auth/auth.tsx [hunk 4]
```diff
@@ -54,14 +55,28 @@ const Auth: React.FC = () => {
...
   );
 
+  const login = useGoogleLogin({
+    onSuccess: ({ code }) => {
+      void dispatch(authActions.loginWithGoogle({ code }));
+    },
+
+    flow: 'auth-code',
+  });
+  const handleGoogleLogin = useCallback(() => login(), [login]);
```
Comment (andanf-e) [inline frontend/src/pages/auth/auth.tsx]: Can't we pass this handler as it's returned from hook?
```suggestion
  const handleGoogleLogin = useGoogleLogin({
    onSuccess: ({ code }) => {
      void dispatch(authActions.loginWithGoogle({ code }));
    },

    flow: 'auth-code',
  });
```

File: backend/src/packages/auth/auth.controller.ts [hunk 5]
```diff
@@ -207,6 +223,17 @@ class AuthController extends Controller {
       payload: await this.authService.resetPassword(options.body),
     };
   }
+
+  private async LoginWithGoogle(
```
Comment (andanf-e) [inline backend/src/packages/auth/auth.controller.ts]:
```suggestion
  private async loginWithGoogle(
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: frontend/src/pages/auth/components/auth-sign-in-button/auth-sign-in-button.tsx (hunk 3 above)
Comment (PablloZz) [inline frontend/src/pages/auth/components/auth-sign-in-button/auth-sign-in-button.tsx]: There is an enum ButtonType for buttons types, maybe it's better to use it

File: frontend/src/pages/auth/components/auth-sign-in-button/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
+.authSignInButton {
+  width: 100%;
+  height: 45px;
+  color: #2e453b;
```
Comment (PablloZz) [inline frontend/src/pages/auth/components/auth-sign-in-button/styles.module.scss]: Please use variable for the color here

File: backend/src/libs/packages/config/config.package.ts [hunk 7]
```diff
@@ -140,6 +140,20 @@ class Config implements IConfig {
...
         },
       },
+      GOOGLE_AUTH: {
+        GOOGLE_CLIENT_ID: {
+          doc: 'Google Auth api client id',
+          format: String,
+          env: 'GOOGLE_CLIENT_ID',
+          default: null,
+        },
+        GOOGLE_CLIENT_SECRET: {
```
Comment (andanf-e) [inline backend/src/libs/packages/config/config.package.ts]: Here we can leave it as it is -- as we already grouped them under `GOOGLE_AUTH`
```suggestion
      GOOGLE_AUTH: {
        CLIENT_ID: {
          doc: 'Google Auth api client id',
          format: String,
          env: 'GOOGLE_CLIENT_ID',
          default: null,
        },
        CLIENT_SECRET: {
```

**Review by andanf-e — DISMISSED** [review]

**Review by NazarKovtun1 — DISMISSED** [review]

**Review by NazarKovtun1 — APPROVED** [review]

**Review by andanf-e — APPROVED** [review]

**Pull request merged**